docs/styled.md

```diff
@@ -16,18 +19,35 @@ function Greeting ({ name }) {
   // blue, 48px, and scaled 2x text
   return <H1 scale={2}>Hello {name}</H1>
 }
+```
 
-// Component
-const H2 = styled(H1)`
-  font-size: ${fontSize * 2/3}px;
-  color: red;
+Component:
+
+```jsx
+import styled from 'react-emotion'
+
+const Heading = styled('h1')`
+  color: blue;
+  font-size: 48px;
+`
+
+const SmallHeading = styled(Heading)`
+  color: black;
+  font-size: 36px;
 `
 
 function Greeting ({ name }) {
-  return <H2>Hello {name}</H2> // red, 32px, and scaled 2x text
+  return (
+    <span>
+      <Heading>Hello {name}</Heading> // blue 48px
+      <SmallHeading>I'm an explainer</SmallHeading> // black 36px
+    </span>
+  )
 }
-
 ```
+
+When styling your custom components, be sure it inherits the appropriate props otherwise it won't propagate the changes. For more info check [issue #366](https://github.com/emotion-js/emotion/issues/366).
```

Comment:
I don't understand this statement. I think it makes more sense to mention this in the extract static doc and explain that composition is not supported in extract static with an example of how to do composition as is mentioned in that issue but also explain that it is extremely limited because the styles aren't actually merged so problems with specificity may happen.

Reply:
I think this is talking about checking props inside the `styled` calls in general, not just for extract. Doesn't really fit here in either case though.

Reply:
I guess the wording wasn't clear enough. Let me explain where i'm trying to get at:

I just started out a typescript project with emotion and still getting a feel for it, and the first custom component I tried to style simply didn't get the new css applied to it.

```jsx
const PaddedComponent = styled(Component)`
  padding: 10px;
`
// padding was being ignored
```

Eventually I figured that was because `Component` wasn't set up to pass through props. 
(It's a typescript project, so we have to deal with interfaces and such.)

Reading the docs there was no reference to that, so this is what this PR is all about. That and making the example a little tidier for good measure.

Reply:
I think the confusion from us comes from the fact that that particular issue is dealing with static extraction and has nothing to do with the example you presented. I'm trying to figure out what context I'm missing but falling short. If you would just remove this section and place it in another PR for discussion we can merge the other work you've done.